### ISSUE TITLE
Various resources

### ISSUE THREAD

**Issue by jan-kleks** [KyleU/solitaire.gg#28]
I have created this ticket so that people can post links to various data resources (images, sounds, cards, solitaire info, whatever) to help you improve solitaire.gg. You can create then separate ticket issues if you find something interesting.

-- if you still need more sound effects [Freesound](http://www.freesound.org/)

-- do you feel nostalgic about [classic Windows solitaire cards](https://www.fastcodesign.com/1671462/a-real-deck-of-cards-that-immortalizes-windows-30-solitaire) too? Many users do, pixelated graphics is very popular now. Of course, you can't use Windows cards, but you could vectorize (well, they're made of simple shapes) standard PySolFC cards which [look very similar](https://mintguide.org/uploads/posts/2017-03/1490629678_linux_mint_002.jpg). Then you can call them in your app "retro/classic", and you get a nice set of cards. Check [this retro black & white set](https://www.waste.org/~oxymoron/cards/) as well, just remove this strange-looking Egyptian card back here.

-- check out [Aisleriot cards](https://github.com/GNOME/aisleriot/tree/master/cards) and [KPatience](https://www.kde.org/applications/games/kpatience/) cards and backgrounds to see if you can find anything useful.

-- you can also try to modify your standard vector card-set, to optionally use only black & white (+red for numbers and symbols) colors, it should look very stylish this way.

**Comment by jan-kleks** [comment]
Have you looked through the [MobyGames card games list](http://www.mobygames.com/game-group/genre-card-tile-game-solitaire-patience) to check if you might be missing any interesting solitaires (or for feature ideas)? 

I have looked around and found a few interesting solitaire card games: 

[Crispy](http://www.mobygames.com/game/crispy) or [On the Edge](http://system-mac.wikia.com/wiki/On_The_Edge)

[ChessCards](http://www.mobygames.com/game/chesscards)

[A Farewell to Kings](https://boardgamegeek.com/boardgame/111398/farewell-kings), [the computer version](https://www.macupdate.com/app/mac/20487/a-farewell-to-kings).

[Matrix Solitaire](http://spoiledflushgames.com/matrix-solitaire/) and there's already a [computer implementation on GitHub](https://github.com/kmiron/Solitaire).

[Sage Solitaire](http://toucharcade.com/2015/09/24/sage-solitaire-review-my-sage-advice-is-to-play-this-game/), also, pay attention to the minimalistic graphics style. When it comes to fancy solitaire apps, check [this one](https://itunes.apple.com/us/app/*solitaire*/id1140586546) too.

[The Seal of Paragon](https://boardgamegeek.com/thread/1049564/2013-solitaire-pp-contest-entry-seal-pragon-contes), this one requires two Jokers.

**Comment by jan-kleks** [comment]
Kenny's free game assets -- really cool looking [cards with minimalistic crown motifs](http://www.kenney.nl/assets/boardgame-pack). Apart from these cards you can also find there some other awesome stuff.

**Comment by jan-kleks** [comment]
Two more nice-looking card designs:

https://opengameart.org/content/playing-card-assets-52-cards-deck-chips

https://opengameart.org/content/playing-cards

**Comment by KyleU** [comment]
I'm looking at each of your issues, planning to address them all. Thanks so much for your interest in making a great solitaire game, your help is definitely appreciated.

**Comment by jan-kleks** [comment]
Thanks! I will be slowly posting new ideas for PySolFC as well: https://github.com/shlomif/PySolFC/issues So you might want to check its issue tracker from time time to time for discussion and new ideas too.

**Comment by jan-kleks** [comment]
You can also think about adding a card-set which would make it easier to play your solitaire by visually impaired people.

Example
![image](https://cloud.githubusercontent.com/assets/15075571/25391507/afd41852-29d6-11e7-8250-69aca39b5d8d.png)

**Comment by KyleU** [comment]
I've got hundreds of new games ready to go, but I need to do a pass through each of them and determine a layout (where the tableau are, etc). It only takes thirty seconds for each of them, but I haven't found time.

**Comment by jan-kleks** [comment]
>I've got hundreds of new games ready to go...

That's awesome! 

Did you know that they're [playing cards in Unicode](http://www.fileformat.info/info/unicode/block/playing_cards/images.htm)? Maybe you could use them for [these miniature layouts that I'm talking about in this comment](https://github.com/KyleU/solitaire.gg/issues/22#issuecomment-293399844) in the form of font characters or images. Or for a simplified card-set (after resizing, of course)? Or in the manual?

What kind of shuffling algorithm you use in your app? Check out [this PySol randomization discussion](https://sourceforge.net/p/pysolfc/discussion/503708/thread/77fec06c/). As well as [this article](https://blog.codinghorror.com/shuffling/), [this one](https://possiblywrong.wordpress.com/2014/12/01/card-shuffling-algorithms-good-and-bad/), [this one (for nice visualization)](http://www.programming-algorithms.net/article/43676/Fisher-Yates-shuffle).

More solitaire ideas:

[Churchill's solitaire](http://www.telegraph.co.uk/technology/2016/01/25/donald-rumsfeld-launches-solitaire-app-inspired-by-winston-churc/)

[Jokers in Solitaire](https://en.wikipedia.org/wiki/Jokers_in_Solitaire)

[Boardwalk solitaire -- quite unusual ](http://boardwalk.pagat.com)

**Comment by jan-kleks** [comment]
Have you got an option to use [Vegas scoring](http://www.chessandpoker.com/vegas-solitaire-game.html) for your solitaires?

**Comment by jan-kleks** [comment]
[Thematic Solitaire Playing Card Games](https://boardgamegeek.com/geeklist/202663/thematic-solitaire-playing-card-games) -- "This list is for solitaire card games with thematic or role-playing elements that use a standard deck of playing cards. The games on this list may include other basic materials such as dice, pen, paper, and counters, provided that the use of these is minimal and a standard deck of playing cards serves as the central or defining component of gameplay."

**Comment by jan-kleks** [comment]
Another interesting solitaire: [Pair Solitaire](http://pairsolitaire.com/)

**Comment by jan-kleks** [comment]
When looking at performance issues, check these [hardware stats](http://hwstats.unity3d.com) (for mobile and PCs) to make sure that solitaire.gg works well on computers/mobile devices with low specs, but which still got significant market share at the same time.

**Comment by jan-kleks** [comment]
Shlomif has created FOSS card games Google Group which you can also use for discussing solitaire.gg matters as well as some other general solitaire stuff: https://groups.google.com/forum/#!forum/foss-card-games

**Comment by jan-kleks** [comment]
I have discovered this beautiful CC-licensed card-set, just remember to use [the "older" version (Web Archive](https://web-beta.archive.org/web/20120606222240/http://www.snap2objects.com/2012/01/free-vector-playing-cards-deck/), and not [the "newer one"](https://creativemarket.com/mauriciodq/12375-Flat-Playing-Cards-Deck). When you put it in the credits, you can always add this Web Archive link just to be safe. From the CC site: "The licensor cannot revoke these [licence] freedoms as long as you follow the license terms."  You can also find these two CC-licensed thematic card-sets by the same designers: [1](http://www.snap2objects.com/2013/06/star-wars-playing-card-deck/) and [2](http://www.snap2objects.com/2012/02/angry-birds-playing-card-deck-and-vector-characters/). They've got [other CC-licensed stuff](http://www.snap2objects.com/category/freebies/) too.

You can also check [Wikimedia Commons](https://commons.wikimedia.org/wiki/Category:SVG_playing_cards), I can see a few simple card backs with geometrical shapes that might be useful.

[Noun Project](https://thenounproject.com/) -- a lot of very stylish icons (they even match the style of your logo) + an option to download them as CC-licensed.

**Comment by jan-kleks** [comment]
How does you hint algorithms compare to those used in PySolFC?

**Comment by jan-kleks** [comment]
[Another very interesting card-set](http://www.freepik.com/free-vector/poker-cards-icons-collection_1045019.htm) which is free even for a commercial use (just read the license, it's neither CC nor GPL).

[This one](http://www.freepik.com/free-vector/cards-deck_707651.htm) does not look too bad either.

**Comment by jan-kleks** [comment]
Have you read this? [Natural Shuffling versus Full Randomization](http://politaire.com/article/intro.html)

**Comment by KyleU** [comment]
I love Politaire - I exchanged emails with his widow when I sought permission to use his game types, he did great work.

At this point, I've got records for 100,000 games. Changing the shuffle algorithm (or random number generator) would invalidate those, I wouldn't be able to say a certain seed is still winnable

**Comment by jan-kleks** [comment]
Talking about Politaire... I have noticed it's got this unique option to use more card suits and to increase the number of colors (up to 8 suits in 4 colors). Have you checked how these options affect solitaires' gameplay?

**Comment by KyleU** [comment]
Yep, started the work for them ( https://github.com/KyleU/solitaire.gg/blob/master/shared/src/main/scala/models/card/Suit.scala#L24 ), but the plan is to use all your images resources, and there aren't many card sets that include tridents and horseshoes ;)

**Comment by jan-kleks** [comment]
Hmm... Probably some simple card-set would be enough for these additional suits at first. Maybe later on some designers will add these new suits to other card-sets...

When it comes to solitaires with non-standard suits, check out these:

1) [Freecell variant with a very interesting twist
](https://www.youtube.com/watch?v=kT7bEJQsVYk)

2) [Golf variant called Barbershop Quintet](http://www.solitairelaboratory.com/golf.html) (BTW, I strongly recommend you read every card-related article on this website, it's a very good resource)

3) [How a larger number of suits affects Aces Up](http://www.solitairelaboratory.com/acesup.html)

4) Klondike variant [1](www.eggrules.com/app/download/6980413/142+Rules.pdf) and [2](http://www.stardeck.com/games/stardeck-klondike-solitaire/)

**Comment by KyleU** [comment]
Wow, the solitaire laboratory guy has thought about solitaire more than you and I combined. Great resource, I'm going to steal a few ideas for settings and options from their work.

**Comment by jan-kleks** [comment]
Have we got this Klondike variant which allows switching from the 3-card deal to 1-card deal? From the Wikipedia article: "Turning three cards at once to the waste with no limit on passes through the deck, but **allowing the player to switch once** to a single pass through the deck one card at a time; after that single pass, however, the player cannot go back to turning three cards at a time and can turn over no more cards from the waste."

Maybe this 3-card deal to 1-card deal switch variant could work for some other solitaires too?

**Comment by jan-kleks** [comment]
[Shadow Solitaire](https://www.shadowsolitaire.com/) -- once again something unusual

**Comment by jan-kleks** [comment]
Check out [this site](http://www.solitairenetwork.com/) too. Take a look especially at these "Slider" games, but you can find other interesting variants as well.

**Comment by jan-kleks** [comment]
I would also like to drag you attention to two things. In the recent years many solitaires were released that I just call **casual solitaires**. These solitaires contain a lot of **chrome**, i.e. a story and theme (for example, medieval, sci-fi) that usually feel tacked-on. However, if one was to remove this chrome, and focus on the core gameplay, they are actually quite fun to play. There are two main types of such casual solitaires:

1) Golf-based, check out [Regency Solitaire](https://www.youtube.com/watch?v=Re7OwSPEqek), [Faerie Solitaire](https://www.youtube.com/watch?v=Jc_FUtiZS5s), and [Forgotten Tales: Day of the Dead](https://www.youtube.com/watch?v=TW-AMwGUrfU) for best examples. You can also read an extensive [write-up on Fairway Solitaire (which seems to be the first casual solitaire of this kind) at Solitaire Laboratory](http://www.solitairelaboratory.com/solitairearcade/fairway.html).

2) Based on matching pairs, check out [Aloha Solitaire](https://www.youtube.com/watch?v=vG1-QklAdIs) and [Solitaire Perfect Match](https://www.youtube.com/watch?v=TblBGnGO320).

What I find really interesting about these games is how much the card layout changes. Not only the cards create a large variety of different shapes, but often covered and uncovered cards are mixed-up on the table. Hmm... I often wonder whether the layouts for such games could be **automatically generated**...

Another matter that I would like to talk about is **power-ups/bonuses**. It's also somewhat connected with these casual solitaires because they have introduced a large portion of these special effects. Things to consider about power-ups:

1) They can make an easy solitaire harder (negative bonuses, e.g. locked cards that can be unlocked only by a certain combination of cards), and a hard one easier (positive effects, e.g. re-shuffle a given pile of cards, remove a particular row of cards).

2) There are different ways power-ups can be implemented: bonuses chosen by the user (or randomized) before starting a game from a pool of power-ups, wild-cards, earning points or cash (see also Vegas scoring) to buy bonuses, etc.

3) Bonus symbols does not have to look too realistic. Instead of a very realistic picture of, for instance, a bomb, you can just have a [silhouette of a bomb](https://thenounproject.com/search/?q=bomb&i=930538). I think it's also a good idea to be conservative about power-up sound effects...

4) Bonuses and power-ups can bring people that don't normally like solitaires (believe me, I have read quite a few user reviews for solitaires with power-ups), and for the users who play mainly standard solitaires, they can be seen as an opportunity to try something different from time to time.

5) Of course, leaderboards and scoring should be separate for games with power-ups.

**Comment by jan-kleks** [comment]
Have you noticed that PySolFC uses not only French cards for its games? Check, for instance:

1) the game called "Imperial Trumps" for a Klondike with German cards (78 cards with special trump cards)

2) "Hex A Klon" -- hexadecimal Klondike  

3) "Samuri" -- like a Klondike with Japanese Hanafuda cards

4) there are also Indian Ganjifa games with over 100 cards in some variants!

It's worth noting that some people also play solitaires (not present in PySolFC) with [Italian](http://www.solitariconlecarte.it/gioca_on_line20.htm) and Spanish cards

![image](https://cloud.githubusercontent.com/assets/15075571/25784957/16270404-3376-11e7-8a93-cb142e250903.png)

Well, in PySolFC you also have this very interesting category called "special games" with games such as memory or Hanoi puzzles which can be played with standard cards, but are not your typical solitaire games. I guess many of card games I have written about in my previous posts here could be used to nicely expand this category. :) 

Another thing I wanted to talk about is an **alternative card-set with just numbers**. I've noticed that when playing Pyramid, I sometimes confuse the values (when adding) of Jacks, Queens, Kings and Aces. It would be easier to play this solitaire (and maybe some other solitaires) if the characters and Aces were represented by numbers instead.

It's also interesting that **one could easily expand the general number of cards by just adding cards with numbers in-between the cards with characters**... Let's say that we have a card-set numbered 1-20, with Jacks for 21, Queens for 22, and Kings for 23, etc.

And the last thing (for today :P) that I keep thinking about when looking at these **extensive lists** **of games** in solitaire.gg and PySolFC is whether these lists **could be reduced** (or maybe it's good as it is). What I mean is whether something like "Spider One Suit" or "Spider Three Deck" should be separate games, or maybe made into some kind of a subcategory of Spider (with variants as an option in the menu, for instance)... Yeah, and when you have such extensive tools for creating new solitaires as in Politaire or PySolFC, which games are worth giving new names and separate entries and which can be left for the users to tweak/create? Just some of my thoughts... 

PS Have you heard about a themed version of Klondike called [SOS Titanic](https://www.youtube.com/watch?v=3IfvAmFQcP4)? Surprisingly, in this case the theme works very well...

**Comment by jan-kleks** [comment]
[Mini SVG Playing Card Set](http://tekeye.uk/svg/mini-svg-playing-card-set) (with quite original Jokers)
[
Free Playing Cards set](https://superdevresources.com/free-playing-cards-set/) -- it does not look like anything special, but it's free so I'm posting it here

[Game-icons.net](http://game-icons.net/) -- many interesting gaming-related icons, maybe you could use them for additional suit symbols, or something else.

**Comment by KyleU** [comment]
I've got all these card resources pulled down locally (with licenses), and I'm building a pipeline to convert them to work with GG. The hares part is isolating a single card from the full-deck layout. I'm (obviously) not the best with graphic design apps, not too sure how to slice them up correctly. Half of them don't even load correctly in Inkscape.

**Comment by KyleU** [comment]
*The hard part, I mean.

**Comment by jan-kleks** [comment]
Maybe ask here: https://opengameart.org/forums/art-discussion

**Comment by jan-kleks** [comment]
BTW, my friend who has been using GIMP and Inkscape for a very long time suggests this software as an alternative: https://affinity.serif.com/ It seems to be really affordable, and he says that he had no problems when using it for commercial projects (as opposed to GIMP and Inkscape).

**Comment by jan-kleks** [comment]
You can also try this: https://designer.io/

**Comment by KyleU** [comment]
I'll make an issue for each card set you've found.  I'm planning on cutting up all the cards so you can mix and match suits, rank font, face card, pip arrangement, and card background. Just a matter of rendering the svgs to a huge raster file and cutting them up.

**Comment by KyleU** [comment]
Thanks for the advice on vector editors, I'll check them out tonight.

**Comment by jan-kleks** [comment]
What about this color picker (click on a color field to evoke it)?: http://jscolor.com

**Comment by jan-kleks** [comment]
@KyleU How are you doing KyleU? Don't worry, I did not forget your awesome project, I'm still watching its development. :) I just wanted to let you know that a new stable version of PySolFC has been released: https://www.reddit.com/r/Python/comments/8cnoou/new_stable_release_of_pysolfc_an_opensource/

And it looks like there are plans for a web version as well.

**Comment by KyleU** [comment]
Hey there, Jan, good to hear from you. Great job on the release, I follow PySolFC - your advocacy really paid off!
